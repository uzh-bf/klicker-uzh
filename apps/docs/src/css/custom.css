@import 'tailwindcss/base';
@import '@uzh-bf/design-system/dist/style.css';

@import 'tailwindcss/components';

@import 'tailwindcss/utilities';

:root {
  @apply font-sans;
  --ifm-color-primary: #dc6027;
  --ifm-color-primary-dark: rgb(33, 175, 144);
  --ifm-color-primary-darker: rgb(31, 165, 136);
  --ifm-color-primary-darkest: rgb(26, 136, 112);
  --ifm-color-primary-light: rgb(70, 203, 174);
  --ifm-color-primary-lighter: rgb(102, 212, 189);
  --ifm-color-primary-lightest: rgb(146, 224, 208);
  --theme-color-primary: #0028a5;
  --theme-color-primary-80: #3353b7;
  --theme-color-primary-60: #667ec9;
  --theme-color-primary-40: #99a9db;
  --theme-color-primary-20: #ccd4ed;
  --theme-color-secondary: #dc6027;
  --theme-color-secondary-80: #e38052;
  --theme-color-secondary-60: #eaa07d;
  --theme-color-secondary-40: #f1bfa9;
  --theme-color-secondary-20: #f8dfd4;
}

html {
  scroll-padding-top: 125px;
  font-size: 16px;
}
