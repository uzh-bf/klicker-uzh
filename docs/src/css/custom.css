:root {
  --ifm-color-primary: #dc6027;
  --ifm-color-primary-dark: rgb(33, 175, 144);
  --ifm-color-primary-darker: rgb(31, 165, 136);
  --ifm-color-primary-darkest: rgb(26, 136, 112);
  --ifm-color-primary-light: rgb(70, 203, 174);
  --ifm-color-primary-lighter: rgb(102, 212, 189);
  --ifm-color-primary-lightest: rgb(146, 224, 208);
}

.border, .border-2, .md\:border {
  --tw-border-opacity: 1;
  border-color: rgb(229 231 235 / var(--tw-border-opacity));
}

.shadow, .shadow-sm, .hover\:shadow, .md\:shadow, .md\:hover\:shadow-lg {
  --tw-ring-offset-shadow: 0 0 #0000;
  --tw-ring-shadow: 0 0 #0000;
  --tw-shadow: 0 0 #0000;
  --tw-shadow-colored: 0 0 #0000;
}

.container {
  width: 100%;
}

@media (min-width: 640px) {
  .container {
    max-width: 640px;
  }
}

@media (min-width: 768px) {
  .container {
    max-width: 768px;
  }
}

@media (min-width: 1024px) {
  .container {
    max-width: 1024px;
  }
}

@media (min-width: 1280px) {
  .container {
    max-width: 1280px;
  }
}

@media (min-width: 1536px) {
  .container {
    max-width: 1536px;
  }
}

.navbar {
  border-bottom: 2px solid #dc6027;
}

.fixed {
  position: fixed;
}

.absolute {
  position: absolute;
}

.relative {
  position: relative;
}

.bottom-4 {
  bottom: 1rem;
}

.bottom-2 {
  bottom: 0.5rem;
}

.bottom-0 {
  bottom: 0px;
}

.z-10 {
  z-index: 10;
}

.order-2 {
  order: 2;
}

.order-1 {
  order: 1;
}

.m-auto {
  margin: auto;
}

.m-2 {
  margin: 0.5rem;
}

.my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.mb-4 {
  margin-bottom: 1rem;
}

.mt-2 {
  margin-top: 0.5rem;
}

.mt-4 {
  margin-top: 1rem;
}

.mb-2 {
  margin-bottom: 0.5rem;
}

.mb-0 {
  margin-bottom: 0px;
}

.mb-10 {
  margin-bottom: 2.5rem;
}

.mr-1 {
  margin-right: 0.25rem;
}

.mt-3 {
  margin-top: 0.75rem;
}

.mb-20 {
  margin-bottom: 5rem;
}

.mt-20 {
  margin-top: 5rem;
}

.ml-1 {
  margin-left: 0.25rem;
}

.mb-8 {
  margin-bottom: 2rem;
}

.mt-8 {
  margin-top: 2rem;
}

.mt-6 {
  margin-top: 1.5rem;
}

.mb-6 {
  margin-bottom: 1.5rem;
}

.mt-10 {
  margin-top: 2.5rem;
}

.mb-12 {
  margin-bottom: 3rem;
}

.mb-3 {
  margin-bottom: 0.75rem;
}

.block {
  display: block;
}

.flex {
  display: flex;
}

.table {
  display: table;
}

.grid {
  display: grid;
}

.hidden {
  display: none;
}

.h-20 {
  height: 5rem;
}

.h-1 {
  height: 0.25rem;
}

.h-5 {
  height: 1.25rem;
}

.h-80 {
  height: 20rem;
}

.max-h-10 {
  max-height: 2.5rem;
}

.max-h-40 {
  max-height: 10rem;
}

.max-h-96 {
  max-height: 24rem;
}

.max-h-80 {
  max-height: 20rem;
}

.w-full {
  width: 100%;
}

.w-36 {
  width: 9rem;
}

.w-44 {
  width: 11rem;
}

.w-28 {
  width: 7rem;
}

.w-52 {
  width: 13rem;
}

.w-1\/2 {
  width: 50%;
}

.w-10 {
  width: 2.5rem;
}

.max-w-lg {
  max-width: 32rem;
}

.max-w-6xl {
  max-width: 72rem;
}

.flex-1 {
  flex: 1 1 0%;
}

.cursor-default {
  cursor: default;
}

.cursor-pointer {
  cursor: pointer;
}

.grid-cols-1 {
  grid-template-columns: repeat(1, minmax(0, 1fr));
}

.grid-cols-2 {
  grid-template-columns: repeat(2, minmax(0, 1fr));
}

.flex-row {
  flex-direction: row;
}

.flex-row-reverse {
  flex-direction: row-reverse;
}

.flex-col {
  flex-direction: column;
}

.flex-wrap {
  flex-wrap: wrap;
}

.items-center {
  align-items: center;
}

.justify-center {
  justify-content: center;
}

.justify-between {
  justify-content: space-between;
}

.gap-4 {
  gap: 1rem;
}

.gap-1 {
  gap: 0.25rem;
}

.gap-8 {
  gap: 2rem;
}

.gap-6 {
  gap: 1.5rem;
}

.self-center {
  align-self: center;
}

.whitespace-nowrap {
  white-space: nowrap;
}

.rounded {
  border-radius: 0.25rem;
}

.rounded-md {
  border-radius: 0.375rem;
}

.rounded-lg {
  border-radius: 0.5rem;
}

.border {
  border-width: 1px;
}

.border-2 {
  border-width: 2px;
}

.border-solid {
  border-style: solid;
}

.border-gray-300 {
  --tw-border-opacity: 1;
  border-color: rgb(209 213 219 / var(--tw-border-opacity));
}

.bg-gray-100 {
  --tw-bg-opacity: 1;
  background-color: rgb(243 244 246 / var(--tw-bg-opacity));
}

.bg-white {
  --tw-bg-opacity: 1;
  background-color: rgb(255 255 255 / var(--tw-bg-opacity));
}

.bg-gray-50 {
  --tw-bg-opacity: 1;
  background-color: rgb(249 250 251 / var(--tw-bg-opacity));
}

.bg-green-200 {
  --tw-bg-opacity: 1;
  background-color: rgb(187 247 208 / var(--tw-bg-opacity));
}

.bg-gradient-to-b {
  background-image: linear-gradient(to bottom, var(--tw-gradient-stops));
}

.bg-gradient-to-r {
  background-image: linear-gradient(to right, var(--tw-gradient-stops));
}

.from-gray-400 {
  --tw-gradient-from: #9ca3af;
  --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to, rgb(156 163 175 / 0));
}

.from-gray-100 {
  --tw-gradient-from: #f3f4f6;
  --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to, rgb(243 244 246 / 0));
}

.via-gray-300 {
  --tw-gradient-stops: var(--tw-gradient-from), #d1d5db, var(--tw-gradient-to, rgb(209 213 219 / 0));
}

.to-transparent {
  --tw-gradient-to: transparent;
}

.to-gray-100 {
  --tw-gradient-to: #f3f4f6;
}

.p-2 {
  padding: 0.5rem;
}

.p-1 {
  padding: 0.25rem;
}

.p-8 {
  padding: 2rem;
}

.p-3 {
  padding: 0.75rem;
}

.p-4 {
  padding: 1rem;
}

.px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.px-10 {
  padding-left: 2.5rem;
  padding-right: 2.5rem;
}

.px-4 {
  padding-left: 1rem;
  padding-right: 1rem;
}

.px-8 {
  padding-left: 2rem;
  padding-right: 2rem;
}

.pl-4 {
  padding-left: 1rem;
}

.pt-20 {
  padding-top: 5rem;
}

.pr-8 {
  padding-right: 2rem;
}

.text-center {
  text-align: center;
}

.text-right {
  text-align: right;
}

.align-text-bottom {
  vertical-align: text-bottom;
}

.font-sans {
  font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
}

.text-sm {
  font-size: 0.875rem;
  line-height: 1.25rem;
}

.text-xs {
  font-size: 0.75rem;
  line-height: 1rem;
}

.text-base {
  font-size: 1rem;
  line-height: 1.5rem;
}

.text-3xl {
  font-size: 1.875rem;
  line-height: 2.25rem;
}

.text-2xl {
  font-size: 1.5rem;
  line-height: 2rem;
}

.text-lg {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.text-xl {
  font-size: 1.25rem;
  line-height: 1.75rem;
}

.font-bold {
  font-weight: 700;
}

.italic {
  font-style: italic;
}

.text-gray-500 {
  --tw-text-opacity: 1;
  color: rgb(107 114 128 / var(--tw-text-opacity));
}

.opacity-50 {
  opacity: 0.5;
}

.shadow {
  --tw-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1), 0 1px 2px -1px rgb(0 0 0 / 0.1);
  --tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color), 0 1px 2px -1px var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
}

.shadow-sm {
  --tw-shadow: 0 1px 2px 0 rgb(0 0 0 / 0.05);
  --tw-shadow-colored: 0 1px 2px 0 var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
}

.transition-opacity {
  transition-property: opacity;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
}

.duration-1000 {
  transition-duration: 1000ms;
}

.ease-out {
  transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
}

.hover\:opacity-100:hover {
  opacity: 1;
}

.hover\:shadow:hover {
  --tw-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1), 0 1px 2px -1px rgb(0 0 0 / 0.1);
  --tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color), 0 1px 2px -1px var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
}

@media (min-width: 640px) {
  .sm\:w-40 {
    width: 10rem;
  }

  .sm\:w-32 {
    width: 8rem;
  }

  .sm\:flex-row {
    flex-direction: row;
  }

  .sm\:justify-start {
    justify-content: flex-start;
  }

  .sm\:bg-gradient-to-b {
    background-image: linear-gradient(to bottom, var(--tw-gradient-stops));
  }

  .sm\:from-gray-400 {
    --tw-gradient-from: #9ca3af;
    --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to, rgb(156 163 175 / 0));
  }

  .sm\:to-transparent {
    --tw-gradient-to: transparent;
  }

  .sm\:text-4xl {
    font-size: 2.25rem;
    line-height: 2.5rem;
  }

  .sm\:text-3xl {
    font-size: 1.875rem;
    line-height: 2.25rem;
  }

  .sm\:text-xl {
    font-size: 1.25rem;
    line-height: 1.75rem;
  }
}

@media (min-width: 768px) {
  .md\:order-1 {
    order: 1;
  }

  .md\:order-2 {
    order: 2;
  }

  .md\:my-auto {
    margin-top: auto;
    margin-bottom: auto;
  }

  .md\:mt-20 {
    margin-top: 5rem;
  }

  .md\:mt-16 {
    margin-top: 4rem;
  }

  .md\:mt-0 {
    margin-top: 0px;
  }

  .md\:block {
    display: block;
  }

  .md\:hidden {
    display: none;
  }

  .md\:h-60 {
    height: 15rem;
  }

  .md\:w-64 {
    width: 16rem;
  }

  .md\:w-52 {
    width: 13rem;
  }

  .md\:max-w-\[450px\] {
    max-width: 450px;
  }

  .md\:flex-initial {
    flex: 0 1 auto;
  }

  .md\:grid-cols-2 {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }

  .md\:flex-row {
    flex-direction: row;
  }

  .md\:items-start {
    align-items: flex-start;
  }

  .md\:gap-8 {
    gap: 2rem;
  }

  .md\:gap-0 {
    gap: 0px;
  }

  .md\:rounded {
    border-radius: 0.25rem;
  }

  .md\:border {
    border-width: 1px;
  }

  .md\:border-solid {
    border-style: solid;
  }

  .md\:p-2 {
    padding: 0.5rem;
  }

  .md\:px-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }

  .md\:py-2 {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
  }

  .md\:px-10 {
    padding-left: 2.5rem;
    padding-right: 2.5rem;
  }

  .md\:text-6xl {
    font-size: 3.75rem;
    line-height: 1;
  }

  .md\:text-lg {
    font-size: 1.125rem;
    line-height: 1.75rem;
  }

  .md\:shadow {
    --tw-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1), 0 1px 2px -1px rgb(0 0 0 / 0.1);
    --tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color), 0 1px 2px -1px var(--tw-shadow-color);
    box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
  }

  .md\:hover\:shadow-lg:hover {
    --tw-shadow: 0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1);
    --tw-shadow-colored: 0 10px 15px -3px var(--tw-shadow-color), 0 4px 6px -4px var(--tw-shadow-color);
    box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
  }
}

@media (min-width: 1024px) {
  .lg\:mb-0 {
    margin-bottom: 0px;
  }

  .lg\:h-80 {
    height: 20rem;
  }

  .lg\:w-1\/3 {
    width: 33.333333%;
  }

  .lg\:grid-cols-2 {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }
}
