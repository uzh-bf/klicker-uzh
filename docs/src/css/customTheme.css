:root {
  --ifm-color-primary-lightest: #3f91e1;
  --ifm-color-primary-lighter: #2281dc;
  --ifm-color-primary-light: #217bd2;
  --ifm-color-primary: #1e70bf;
  --ifm-color-primary-dark: #1b65ac;
  --ifm-color-primary-darker: #195fa2;
  --ifm-color-primary-darkest: #154e86;
}

* {
  font-family: $myFont !important;
}

/* custom header color */
.fixedHeaderContainer {
  background-color: $customHeaderColor;
}

.navigationSlider .slidingNav ul li a:focus,
.navigationSlider .slidingNav ul li a:hover,
.navigationSlider .slidingNav ul li.siteNavItemActive a,
.navigationSlider .slidingNav ul li.siteNavGroupActive a {
  background-color: $customHeaderColor;
}
/* custom header color */

.mainContainer .wrapper .post h1,
.mainContainer .wrapper .post h2,
.mainContainer .wrapper .post h3 {
  font-weight: 500;
}

.mainContainer .wrapper .post h3 {
  color: black;
  margin: 20px 0 0 0;
  padding: 0px;
}

h2.projectTitle {
  font-size: 3rem;
  line-height: 3rem;
}

h2.projectTitle span {
  vertical-align: top;
  font-size: 1.5rem;
  line-height: 1.5rem;
}

h2,
h2.projectTitle {
  font-weight: 500;
}

.lightBackground {
  background-color: #7cb8e433;
}

img[alt="Student View Mobile"] {
  max-width: 400px;
}

footer.nav-footer {
  background-color: white;
}

footer.nav-footer .copyright {
  color: grey;
}
