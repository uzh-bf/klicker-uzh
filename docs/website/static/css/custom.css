* {
  font-family: $myFont !important;
}

/* custom header color */
.fixedHeaderContainer {
  background-color: $customHeaderColor;
}

.navigationSlider .slidingNav ul li a:focus,
.navigationSlider .slidingNav ul li a:hover,
.navigationSlider .slidingNav ul li.siteNavItemActive a,
.navigationSlider .slidingNav ul li.siteNavGroupActive a {
  background-color: $customHeaderColor;
}
/* custom header color */

.mainContainer .wrapper .post h1,
.mainContainer .wrapper .post h2,
.mainContainer .wrapper .post h3 {
  font-weight: 500;
}

.mainContainer .wrapper .post h3 {
  color: black;
  margin: 20px 0 0 0;
  padding: 0px;
}

h2.projectTitle {
  font-size: 3rem;
  line-height: 3rem;
}

h2.projectTitle span {
  vertical-align: top;
  font-size: 1.5rem;
  line-height: 1.5rem;
}

h2,
h2.projectTitle {
  font-weight: 500;
}

.lightBackground {
  background-color: #7cb8e433;
}

img[alt='Student View Mobile'] {
  max-width: 400px;
}

footer.nav-footer {
  background-color: white;
}

footer.nav-footer .copyright {
  color: grey;
}
