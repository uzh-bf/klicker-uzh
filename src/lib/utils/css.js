// @flow

/* eslint-disable import/prefer-default-export */

import React from 'react'

function createLinks(links: Array<string> = []): Array<*> {
  return links.map(link => (
    <link
      key={link}
      rel="stylesheet"
      href={
        link.substr(0, 4) === 'http' ? (
          link
        ) : (
          `https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.11/components/${link}.min.css`
        )
      }
    />
  ))
}

export { createLinks }
