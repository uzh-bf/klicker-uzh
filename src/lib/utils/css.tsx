/* eslint-disable import/prefer-default-export */

import React from 'react'

export function createLinks(links: string[] = []): React.ReactElement[] {
  return links.map(link => <link href={link} key={link} rel="stylesheet" />)
}

