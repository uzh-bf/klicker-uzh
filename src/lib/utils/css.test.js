import { createLinks } from './css'

describe('createLinks', () => {
  it('works with full urls', () => {
    expect(createLinks(['http://www.google.com'])).toMatchSnapshot()
  })

  it('works with semantic components', () => {
    expect(createLinks(['dropdown', 'header', 'menu'])).toMatchSnapshot()
  })

  it('works with combinations of both', () => {
    expect(createLinks(['http://www.google.com', 'header'])).toMatchSnapshot()
  })
})
