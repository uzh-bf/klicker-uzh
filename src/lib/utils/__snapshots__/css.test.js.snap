// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createLinks works with full urls 1`] = `
Array [
  <link
    href="http://www.google.com"
    rel="stylesheet"
  />,
]
`;

exports[`createLinks works with semantic components 1`] = `
Array [
  <link
    href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.11/components/dropdown.min.css"
    rel="stylesheet"
  />,
  <link
    href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.11/components/header.min.css"
    rel="stylesheet"
  />,
  <link
    href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.11/components/menu.min.css"
    rel="stylesheet"
  />,
]
`;
